src/resqml2_0_1/RockFluidOrganizationInterpretation.cpp

```diff
@@ -0,0 +1,98 @@
+/*-----------------------------------------------------------------------
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"; you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-----------------------------------------------------------------------*/
+#include "resqml2_0_1/RockFluidOrganizationInterpretation.h"
+#include <algorithm>
+
+#include "tools/Misc.h"
+
+using namespace std;
+using namespace epc;
+using namespace RESQML2_0_1_NS;
+using namespace gsoap_resqml2_0_1;
+
+const char* RockFluidOrganizationInterpretation::XML_TAG = "RockFluidOrganizationInterpretation";
+
+RockFluidOrganizationInterpretation::RockFluidOrganizationInterpretation(OrganizationFeature * orgFeat, const std::string & guid, const std::string & title)
+{
+	if (orgFeat == nullptr) {
+		throw invalid_argument("The interpreted organization feature cannot be null.");
+	}
+	if (!orgFeat->isPartial() && orgFeat->getKind() != gsoap_resqml2_0_1::resqml2__OrganizationKind__fluid) {
+		throw invalid_argument("The kind of the organization feature is not a fluid organization.");
+	}
+
+	gsoapProxy2_0_1 = soap_new_resqml2__obj_USCORERockFluidOrganizationInterpretation(orgFeat->getGsoapContext(), 1);
+	_resqml2__RockFluidOrganizationInterpretation* rfoi = static_cast<_resqml2__RockFluidOrganizationInterpretation*>(gsoapProxy2_0_1);
+	rfoi->RockFluidUnitIndex = soap_new_resqml2__RockFluidUnitInterpretationIndex(orgFeat->getGsoapContext(), 1);
+	// No need to initialize index since it is a bug : http://docs.energistics.org/#RESQML/RESQML_TOPICS/RESQML-500-106-0-R-sv2010.html
+
```

Comment:
Probably here, a link to a geologicUnitInterpretation is needed. It should even be a its RockFluidUnitInterpretation specialization.

Reply:
the class geologicUnitInterpretation does not exist. Should I create it ?

Reply:
Up to you.
That's better long term but it is not needed for now (it can definitely be part of another PR). It is quicker not to create GeologicUnitInterpretation.
Having/Creating a RockFluidUnitInterpretation is the most important in my opinion. It can derive directly from AbstractFeatureInterpretation as StratigraphicUnitInterpretation currently does.